internal/characters/barbara/barbara.go

```diff
@@ -0,0 +1,118 @@
+package barbara
+
+import (
+	"github.com/genshinsim/gcsim/pkg/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+)
+
+func init() {
+	core.RegisterCharFunc(keys.Barbara, NewChar)
+}
+
+type char struct {
+	*character.Tmpl
+	stacks     int
+	c6icd      int
+	skillInitF int
+	// burstBuffExpiry   int
+}
+
+func NewChar(s *core.Core, p core.CharacterProfile) (core.Character, error) {
+	c := char{}
+	t, err := character.NewTemplateChar(s, p)
+	if err != nil {
+		return nil, err
+	}
+	c.Tmpl = t
+	c.Energy = 80
+	c.EnergyMax = 80
+	c.Weapon.Class = core.WeaponClassCatalyst
+	c.BurstCon = 3
+	c.SkillCon = 5
+	c.NormalHitNum = 4
+
+	c.a1()
+
+	if c.Base.Cons >= 1 {
+		c.c1(1)
+	}
+	if c.Base.Cons >= 2 {
+		c.c2()
+	}
+	if c.Base.Cons >= 6 {
+		c.c6()
+	}
+	return &c, nil
+}
+
+func (c *char) a1() {
+	c.Core.AddStamMod(func(a core.ActionType) (float64, bool) { // @srl does this activate for the active char?
+		if c.Core.Status.Duration("barbskill") >= 0 {
+			return -0.12, false
+		}
+		return 0, false
+	})
+}
+
+func (c *char) c2() {
```

Comment:
for this take a look at how raiden's E is implemented. you want to add this mod onto every single character

also by design the Amount function returns 2 values. first is the stat slice that has the changes to stat to be added to the snapshot. second is a bool stating if the buff applies or not

so if the bool is false, the first slice is ignored all together. so it's fine to just have val[core.HydroP] = 0.15 and never change that. and when it doesnt apply just return nil, false

Reply:
do you mean the burst? I think I changed it to what it is correctly but will need another review